lib/java/AdapterLibrary/src/main/kotlin/com/vmware/aria/operations/Data.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.aria.operations
+
+/**
+ * Class representing a Metric Data Point.
+ * Metrics are numeric values that represent data at a particular point in time.
+ * These are stored as time series data.
+ *
+ * Examples:
+ *    CPU Utilization
+ *    Disk Capacity
+ *    Current User Session Count
+ *    Cumulative Data Received
+ *
+ * @param key A string identifying metric.
+ * @param value The value of the metric. Note: VMware Aria Operations treats
+ * all numeric values as a [Double].
+ * @param timestamp Time in milliseconds since the Epoch when this metric value was
+ * recorded. Defaults to the current time.
+ */
+class Metric @JvmOverloads constructor(
+    val key: String,
+    val value: Number,
+    val timestamp: Long = System.currentTimeMillis()
+) {
+
+    /**
+     * This is identical to calling 'value' directly. It is included to make working
+     * with metrics and properties similar.
+     *
+     * @return the value of the metric as a double.
+     */
+    val doubleValue
+        get() = value.toDouble()
+    /**
+     * Get a JSON representation of this Metric.
+     * This method returns a JSON representation of this Metric in the format required by VMware Aria Operations.
+     * @return The JSON representation of this Metric, as a Map
+     */
+    val json: Map<String, Any>
+        get() = mapOf(
+            "key" to key,
+            "numberValue" to doubleValue,
+            "timestamp" to timestamp
+        )
+}
+
+sealed class PropertyValueType
+class StringValue(val string: String) : PropertyValueType()
+class DoubleValue(val double: Double) : PropertyValueType()
+
+/**
+ * Class representing a Property value.
+ *
+ * A Property is a value, usually a string, that will change infrequently or not at all.
+ * Only the current value is important (i.e., a graph doesn't make sense).
```

Comment:
I don't follow the sentence about importance. The historical value of a property might be important (e.g., a config change causes an outage or performance issue).